### ISSUE TITLE
Dart2JS cmdline usage info is outdated

### ISSUE THREAD

**Issue by lexaknyazev** [dart-lang/sdk#30114]
List of Dart2JS command-line options is [hard-coded](https://github.com/dart-lang/sdk/blob/3b05eb95530b1684a3fbcc69bda1784db393b966/pkg/compiler/lib/src/dart2js.dart#L671) and doesn't reflect its current state (e.g. it lacks `--trust-type-annotations`, but mentions already removed `dart:dom` library). Maybe, usage info should be auto-generated from the actual list of recognized flags?

This issue (or more specific ones about certain options) was brought up several times before:
https://github.com/dart-lang/sdk/issues/4846
https://github.com/dart-lang/sdk/issues/11113
https://github.com/dart-lang/sdk/issues/18773
https://github.com/dart-lang/sdk/issues/21055
https://github.com/dart-lang/sdk/issues/28886

**Comment by sigmundch** [comment]
Thanks again for bringing this up. 

Kathy just uploaded more details on --trust-type-information in https://webdev.dartlang.org/tools/dart2js

I just created a CL to update the documentation from the compiler directly as well.

**Comment by sigmundch** [comment]
I agree it would be nice if this is auto-generated, but we need to change how we manage options internally first before we can do that. For now, the least we can do is update the information

**Comment by sigmundch** [comment]
Just to put here some of the ideas we have in mind:
 - introduce options as an object so we can easily collect them and aggregate the help messages for multiple options
 - annotation options with a "category" so it is easy to divide them in groups:
   - essential options show up with -h
   - non-essential option show up with -h -v
   - non-essential options can be subdivided in groups. Groups that we know of so far:
     - customization: tune the compiler, like `--csp` or `--no-source-maps`
     - advanced: features that should be used with care (`--trust-type-annotations`, `--library-root`)
     - early-access: features that are likely going to be part of dart2js, but are not well tested (e.g. new-source-info, etc)
     - experimental: features that are expected to change, break, or get deleted.
